metavalidator/all_test.go

```diff
@@ -0,0 +1,92 @@
+package metavalidator_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/metavalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+)
+
+func TestAllValidator(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		val                    attr.Value
+		valueValidators        []tfsdk.AttributeValidator
+		expectError            bool
+		inspectDiags           bool
+		expectedValidatorDiags diag.Diagnostics
+	}
+	tests := map[string]testCase{
+		"Type mismatch": {
+			val: types.Int64{Value: 12},
+			valueValidators: []tfsdk.AttributeValidator{
+				stringvalidator.LengthAtLeast(3),
+				stringvalidator.LengthAtLeast(5),
+			},
+			expectError: true,
+			// We can't test the diags returned as they are in the /internal/reflect pkg.
```

Comment:
Aye this has bitten me in the past as well -- I think we should introduce an `Equal()` method on `diag.Diagnostics` so go-cmp can use the underlying `(diag.Diagnostic).Equal()` equality checking, rather than it needing to be Go type specific. I'll raise an issue upstream.